diff -urp /usr/sbin/automysqlbackup /usr/sbin/automysqlbackup
--- /usr/sbin/automysqlbackup	2014-07-23 16:13:44.000000000 +0800
+++ /usr/sbin/automysqlbackup	2018-03-14 00:24:16.485982724 +0800
@@ -433,7 +433,7 @@ dbdump () {
 	if [ -z "${USERNAME}" -o -z "${PASSWORD}" ] ; then
 		mysqldump --defaults-file=/etc/mysql/debian.cnf $NEWOPT $1 > $2
 	else
-		mysqldump --user=$USERNAME --password=$PASSWORD --host=$DBHOST $NEWOPT $1 > $2
+		MYSQL_PWD=$PASSWORD mysqldump --user=$USERNAME --host=$DBHOST $NEWOPT $1 > $2
 	fi
 	return 0
 }
@@ -499,7 +499,7 @@ if [ "$DBNAMES" = "all" ]; then
 	if [ -z "${USERNAME}" -o -z "${PASSWORD}" ] ; then
         DBNAMES="`mysql --defaults-file=/etc/mysql/debian.cnf --batch --skip-column-names -e "show databases"| sed 's/ /%/g'`"
 	else
-        DBNAMES="`mysql --user=$USERNAME --password=$PASSWORD --host=$DBHOST --batch --skip-column-names -e "show databases"| sed 's/ /%/g'`"
+        DBNAMES="`MYSQL_PWD=$PASSWORD mysql --user=$USERNAME --host=$DBHOST --batch --skip-column-names -e "show databases"| sed 's/ /%/g'`"
 	fi
 
 	# If DBs are excluded
